app/Http/Controllers/AvatarController.php

```diff
@@ -11,48 +11,28 @@ class AvatarController extends Controller
     public function profile()
     {
         // Retrieve the currently authenticated user
-        $user = Auth::user();
-        if(Auth::check()){
-            $account = Account::find($user->user_ID);
-            return view('userprofile', compact('account'));
+        
+        $account = Auth::user();;
+        return view('userprofile', compact('account'));
 
-        } else{
-            return view('please_login');
-        }
-}
-
-    // app/Http/Controllers/AccountController.php
-
-
-// ...
-
-public function updateAvatar(Request $request)
-{
-    // Validate and save the new profile picture (avatar)
-    $request->validate([
-        'avatar' => 'required|image|mimes:jpeg,png,jpg,gif|max:2048', // Example validation rules
-    ]);
-    $user = Auth::user();
-
-    Account::find($user->user_ID)->avatar = $request->avatar;
-    
-    Account::find($user->user_ID)->save();
+       
+    }
+    public function update(Request $request)
+    {
+        // Retrieve the currently authenticated user
+        $description = $request->query('description');
+        $account =  Auth::user();;
+        $data = $request->all();
+        $account->name = $data['name'];
+        $account->email = $data['email'];
+        $account->avatar = $data['avatar'];
```

Comment:
this might be a file upload. take it into consideration

General:
if you are working with arrays, you must always assume that the key might not exist.

instead of $data['name'], use $data['name'] ?? ''

Specific

Specific to you case here, don't use $data = $request->all().

Instead use $account->name = $request->post->get('name') ( i believe this was the correct method call)